components/reddit/sources/new-comments-by-user/new-comments-by-user.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-comments-by-user",
+  name: "New comments by user",
+  description: "Emits an event each time a user posts a new comment.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    username: { propDefinition: [reddit, "username"] },
+    context: {
+      type: "integer",
+      label: "context",
+      description: "an integer between 2 and 10",
+    },
+    t: { propDefinition: [reddit, "t"] },
+    sr_detail: { propDefinition: [reddit, "sr_detail"] },
+    ...common.props,
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      let before = null;
+      try {
+        var reddit_comments = await this.reddit.getNewUserComments(
+          null,
+          this.username,
+          this.context,
+          this.t,
+          this.sr_detail,
+          10
+        );
+      } catch (err) {
+        if (common.methods.did4xxErrorOccur(err)) {
```

Comment:
Since we include `...common.methods` in the component methods, we can refer to a method from common.js like this: `this.did4xxErrorOccurs(err)` or `this.wereThingsPulled(reddit_comments)` just like when the method is written in the component itself.

Reply:
Got it. I made the change. Not sure what happened but when I try `this` it didn't work. I guess it was a transient issue with the coding.